src/librustc_borrowck/diagnostics.rs

```diff
@@ -286,6 +286,66 @@ You can read more about cell types in the API documentation:
 https://doc.rust-lang.org/std/cell/
 "##,
 
+E0389: r##"
+An attempt was made to mutate data using a non-mutable reference. This
+commonly occurs when attempting to assign to a reference of type
```

Comment:
I think your second sentence is a bit unwieldly. What do you think of: "This commonly occurs when attempting to assign to a non-mutable reference of a mutable reference"?